cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -224,7 +224,8 @@ protected void startUp() throws Exception {
       final URI jobFile = context.isPySpark() ? getPySparkScript(tempDir) : createJobJar(tempDir);
       List<File> extraPySparkFiles = new ArrayList<>();
 
-      String metricsConfPath;
+      boolean sparkMetricsEnabled = cConf.getBoolean(Constants.Metrics.SPARK_METRICS_ENABLED);
```

Comment:
would it be better to give it a default value here too ? using the getBoolean(String, boolean) method ?

Reply:
We usually don't do defaults in get calls and rely on cdap-defaults.xml,. This ensures we have defaults in one place and they don't get out of sync

Reply:
IMO, it's better to define default values in cdap-default as it's been done already.

Reply:
sg , thanks for the explanation :)